topic-operator/src/main/java/io/strimzi/operator/topic/Config.java

```diff
@@ -159,6 +176,24 @@ private Value(String key, Type<? extends T> type, boolean required) {
     /** The endpoint identification algorithm used by clients to validate server host name. The default value is https. Clients including client connections created by the broker for inter-broker communication verify that the broker host name matches the host name in the broker’s certificate. Disable server host name verification by setting to an empty string.**/
     public static final Value<String> TLS_SSL_ENDPOINT_IDENTIFICATION_ALGORITHM = new Value<>(TC_TLS_SSL_ENDPOINT_IDENTIFICATION_ALGORITHM, STRING, "HTTPS");
 
+    /**
+     * The store topic for the Kafka Streams based TopicStore
+     */
+    public static final Value<String> STORE_TOPIC = new Value<>(TC_STORE_TOPIC, STRING, "__strimzi_store_topic");
+    /** The store name for the Kafka Streams based TopicStore */
+    public static final Value<String> STORE_NAME = new Value<>(TC_STORE_NAME, STRING, "__strimzi_topic_store");
```

Comment:
I am fine with using the `__` prefix for an internal topic (see the store topic) but why using this kind of notation for something that isn't a topic, so the store itself? Isn't it enough just "topics-store" or something like that? While the topic is something visible to the user (the TO will create a `KafkaTopic` resource named `__strimzi_store_topic`), the store is not. @tombentley @alesj any thoughts? My confusion comes from using "store" and "topic" too many times where the order they are used matters.

Reply:
Streams create (log) topics based on the store name.
This way you can immediately tell this is an internal topic.
(although I guess they already have "log" in the name, which tells you it's a Streams log topic ...)

Reply:
Right, I forgot the log topic ... can you double-check the final name so that we can figure out a better name for the store?

Reply:
<application.id>-<store.name>-changelog

Reply:
so it's going to be `strimzi-topic-store-__strimzi_topic_store-changelog` ... I really think it's a crazy one!! :-D
The `__` doesn't make sense anymore because it's not at the beginning of the topic name so it doesn't highlight it as an internal topic, then again strimzi, topic and store used too much :-) Isn't it possible to override the changelog topic name?

Reply:
> Isn't it possible to override the changelog topic name?

No idea ...

Reply:
So after some investigation, it turned out that it's not possible to override the name completely. The internal Kafka Streams topics have always `-changelog` or `-repartition` suffixes for topics related to store and repartitioning.
So we can just "play" with <application.id> and <store.name> as for the format you pasted.
First, if we want the final topic starting with `__` then we need the application id starting with `__`.
Maybe something like:

application.id = "__strimzi-topic-operator-kstreams"
store.name = "topic-store"

so changelog topic will be `__strimzi-topic-operator-kstreams-topic-store-changelog`

Tbh I run out of ideas :-)

Reply:
OK, I'll put this values into Config ...